tests/test_all.py

```diff
@@ -77,3 +77,14 @@ def test_filter_by_non_indexed_field_should_fail(self):
                 db=self.db,
                 status='active',
             ))
+
+    def test_all_iter(self):
+        names_in_expected_order = ['Test name', 'Test name2', 'ZTest name']
+
+        async def _test_loop():
+            count = 0
+            async for x in TestUser.all_iter(db=self.db, order_by='name'):
```

Comment:
@mflaxman  We cannot have return and yield at the same time  in a method that is the reason for separate  methods.